src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.cs

```diff
@@ -446,22 +446,73 @@ internal void ValidateCanBeUsedForMetadataSerialization()
         /// <exception cref="InvalidOperationException">
         /// The <see cref="JsonTypeInfo"/> instance has been locked for further modification.
         /// </exception>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Specified an invalid <see cref="JsonNumberHandling"/> value.
+        /// </exception>
         /// <remarks>
         /// For contracts originating from <see cref="DefaultJsonTypeInfoResolver"/> or <see cref="JsonSerializerContext"/>,
-        /// the value of this callback will be mapped from any <see cref="JsonNumberHandling"/> annotations.
+        /// the value of this callback will be mapped from any <see cref="JsonNumberHandlingAttribute"/> annotations.
         /// </remarks>
         public JsonNumberHandling? NumberHandling
         {
             get => _numberHandling;
             set
             {
                 VerifyMutable();
+
+                if (value is not null && !JsonSerializer.IsValidNumberHandlingValue(value.Value))
+                {
+                    throw new ArgumentOutOfRangeException(nameof(value));
+                }
```

Comment:
should validation be happening in configure? no strong opinion but we have most of it there already

Reply:
We should be checking if the provided values are out of range, similar to how the `JsonSerializerOptions` setters do this. `Configure` does more elaborate validation, like checking if combinations of individual valid settings can be used together.